qiskit_nature/utils/linalg.py

```diff
@@ -179,3 +179,136 @@ def _swap_columns(matrix: np.ndarray, i: int, j: int) -> None:
     column_i = matrix[:, i].copy()
     column_j = matrix[:, j].copy()
     matrix[:, i], matrix[:, j] = column_j, column_i
+
+
+def modified_cholesky(
+    mat: np.ndarray, *, error_threshold: float = 1e-8, max_vecs: int | None = None
```

Comment:
How will this currently handle say a `sparse.SparseArray` object? It would be nice to test that and (if it works out of the box) add a unittest for this case to ensure that it stays that way.

Reply:
```
    @data(4, 5)
    def test_modified_cholesky_sparse(self, dim: int):
        """Test modified Cholesky decomposition on a random tensor using sparse."""
        rng = np.random.default_rng(4640)
        # construct a random positive definite matrix
        unitary = np.array(random_unitary(dim, seed=rng))
        eigs = rng.uniform(size=dim)
        mat = sparse.COO(unitary @ np.diag(eigs) @ unitary.T.conj())
>       cholesky_vecs = modified_cholesky(mat)

test/utils/test_linalg.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
qiskit_nature/utils/linalg.py:234: in modified_cholesky
    max_error_index = np.argmax(errors)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<COO: shape=(4,), dtype=float64, nnz=4, fill_value=0.0>,), kwargs = {}, relevant_args = (<COO: shape=(4,), dtype=float64, nnz=4, fill_value=0.0>, None)

>   ???
E   TypeError: no implementation found for 'numpy.argmax' on types that implement __array_function__: [<class 'sparse._coo.core.COO'>]

<__array_function__ internals>:180: TypeError
```

Reply:
Hm, that is unexpected. I wonder whether upgrading to numpy >= 1.24 will fix this. But that requires `numba` to release first which I expect to happen in the near future. I will keep an eye on this, but I don't think this blocks this PR at this time. But let's leave this thread open.